### PR TITLE
Feature/improve sha token

### LINKED ISSUE #5: It would be great if IDs were alphanumeric and not easy to be guessed
Hey, 

I'm not a big fan of using numeric ids. That is: 

- https://skunk.fastruby.io/1
- https://skunk.fastruby.io/2
- https://skunk.fastruby.io/3
- etc...

I think it would be great if we could automatically generate a slug that is alphanumeric. For example: 

- https://skunk.fastruby.io/d530e56
- https://skunk.fastruby.io/a32f65ab
- etc... 

That way, users could have some privacy in terms of sharing their skunk information. 

URLs would continue to be publicly accessible, but unwanted visitors would have a harder time guessing the slug. 😄 

What do you think?

### PR DESCRIPTION
**Pull request by etagwerker**
Hey, 

This PR adds a way to generate URLs that are harder to be guessed. It fixes #5. 

This PR depends on #18, so please review that one first. 😺 

Please check it out. 

Thanks!

### CODE REVIEW

File: test/integration/routing_test.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require 'test_helper'
 
 class RoutingTest < ActionDispatch::IntegrationTest
```
Comment (arielj) [inline test/integration/routing_test.rb:3]: Just a suggestion, the name an location of this test may be confusing compared to the rails guides. They suggest that routing tests should go inside test/controller and should use assertions like `assert_routing "/#{report.slug}", controller: 'reports', action: 'show'`.

https://guides.rubyonrails.org/testing.html#testing-routes

**Review by arielj — CHANGES_REQUESTED** [review]
The only required change would be the index in the table. The test name/location comment is not really required.

File: test/integration/routing_test.rb (hunk 1 above)
Comment (etagwerker) [reply test/integration/routing_test.rb]: @arielj Good point. I just renamed the class name. This file is not really a routing test but more of an integration test.

File: db/migrate/20201201212814_add_slug_to_reports.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddSlugToReports < ActiveRecord::Migration[6.0]
+  def change
+    add_column :reports, :slug, :string
```
[the remark this answers is not in the document]
Comment (etagwerker) [reply db/migrate/20201201212814_add_slug_to_reports.rb]: @arielj Good catch, thanks! 👍

**Comment by etagwerker** [comment]
@arielj I just finished addressing your comments.

**Review by arielj — APPROVED** [review]
LGTM!

**Pull request merged**